### DISCUSSION TITLE
How accurate are the docs regarding configuration?

### DISCUSSION THREAD

**Discussion by wsw70** [custom-components/pyscript#362] [Q&A]
I am getting back to more automation and more heated family discussions when something does not work and I've been re-reading the docs regarding configuration (https://hacs-pyscript.readthedocs.io/en/latest/reference.html#configuration)

How accurate are they?

I had a hard look at my scripts mess and realized that none are under `apps`, there is no mention of `apps` in `pyscript/config.yaml` etc. But they do work.

---

As a side note, I think it would be awesome to have a configuration file per "app" (a directory) to avoid dragging a single configuration file at the top level.

**Comment by craigbarratt** [comment]
They should be accurate - if not please submit a PR or describe any errors so they can be fixed.

Yes, you don't have to use `apps`.  All the configuration settings are available in `pyscript.config`.  The `apps` structure is to allow better partitioning and encapsulation of applications, to make it easier to share them etc.  You could also put each application's `yaml` settings in the `pyscript/apps/MYAPP` directory, and include it from the main config file.

See also [this section](https://hacs-pyscript.readthedocs.io/en/latest/reference.html#accessing-yaml-configuration) of the docs.

**Reply by wsw70** [reply]
Thank you. Is the word `apps` special in some sense? I am trying to understand what I can do when putting apps in `apps` vs having them elsewhere.

**Reply by craigbarratt** [reply]
Yes, the word `apps` is special.  Directories and files below the `apps` directory, and configuration parameters below `apps` are treated differently, as described in the documentation.
